nucliadb/src/nucliadb/learning_proxy.py

```diff
@@ -418,3 +416,69 @@ def _handle_request(self, *args: Any, **kwargs: Any) -> httpx.Response:
             return getattr(self, method)(*args, **kwargs)
         else:
             return self._response()
+
+
+class LearningConfigService(ABC):
+    @abstractmethod
+    async def get_configuration(self, kbid: str) -> Optional[LearningConfiguration]: ...
+
+    @abstractmethod
+    async def set_configuration(self, kbid: str, config: dict[str, Any]) -> LearningConfiguration: ...
+
+    @abstractmethod
+    async def delete_configuration(self, kbid: str) -> None: ...
+
+
+class ProxiedLearningConfig(LearningConfigService):
+    async def get_configuration(self, kbid: str) -> Optional[LearningConfiguration]:
+        async with self._client() as client:
+            resp = await client.get(f"config/{kbid}")
+            try:
+                resp.raise_for_status()
+            except httpx.HTTPStatusError as err:
+                if err.response.status_code == 404:
+                    return None
+                raise
+            return LearningConfiguration.model_validate(resp.json())
+
+    async def set_configuration(self, kbid: str, config: dict[str, Any]) -> LearningConfiguration:
+        async with self._client() as client:
+            resp = await client.post(f"config/{kbid}", json=config)
+            resp.raise_for_status()
+            return LearningConfiguration.model_validate(resp.json())
+
+    async def delete_configuration(self, kbid: str) -> None:
+        async with self._client() as client:
+            resp = await client.delete(f"config/{kbid}")
+            resp.raise_for_status()
+
+    @contextlib.asynccontextmanager
+    async def _client(self) -> AsyncIterator[httpx.AsyncClient]:
+        async with httpx.AsyncClient(
+            base_url=get_base_url(LearningService.CONFIG),
+            headers=get_auth_headers(),
+        ) as client:
+            yield client
+
+
+class InMemoryLearningConfig(LearningConfigService):
+    def __init__(self):
+        self.in_memory_configs = {}
+
+    async def get_configuration(self, kbid: str) -> Optional[LearningConfiguration]:
+        return self.in_memory_configs.get(kbid, None)
+
+    async def set_configuration(self, kbid: str, config: dict[str, Any]) -> LearningConfiguration:
+        parsed_config = LearningConfiguration.model_validate(config)  #
+        self.in_memory_configs[kbid] = parsed_config
+        return parsed_config
+
+    async def delete_configuration(self, kbid: str) -> None:
+        self.in_memory_configs.pop(kbid, None)
```

Comment:
This is not stored as part of the global state anywhere right?
So from one request to the next, the config won't be "persisted" even if the process hasn't restarted

Reply:
Ups! Changed for a global LRU